src/DetectorConstruction.cc

```diff
@@ -31,18 +29,12 @@ G4VPhysicalVolume* DetectorConstruction::Construct() {
     // Reading the geometry
     TString geometryFile = restG4Metadata->Get_GDML_Filename();
 
-    // char originDirectory[255];
-    // sprintf( originDirectory, "%s", get_current_dir_name() );
-
-    // char buffer[255];
-    // sprintf( buffer, "%s", (char *) restG4Metadata->GetGeometryPath().Data() );
-    // chdir( buffer );
     char originDirectory[256];
     sprintf(originDirectory, "%s", getenv("PWD"));
```

Comment:
The use of environmental variables in the code is not a good practice in general.
In this particular case you can use "std::filesystem::current_path()".

Reply:
That is only a feature of C++17 if I'm correct, sadly

Reply:
Indeed, another reason to move to C++17